### PR TITLE
[Java][Native] Support oneOf/anyOf schemas

### PR DESCRIPTION
**Pull request by bgong-mdsol**
### Description of the PR
This PR includes
- Add oneOf and anyOf implementation (based on java jersey2 approach)
- re-generate sample projects of native/asyncNative
- add a sample project for OneOf

@wing328 @sebastien-rosset @cbornet @jeff9finger @bbdouglas @sreeshas @jfiala @lukoyanov @karismann @Zomzog @lwlee2608 @bkabrda @jfeltesse-mdsol @jcarres-mdsol

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/Java/libraries/native/anyof_model.mustache [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                // discriminate the anyOf schemas.
+                // Get the discriminator mapping value to get the class.
+                deserialized = tree.traverse(jp.getCodec()).readValueAs(cls);
+                {{classname}} ret = new {{classname}}();
+                ret.setActualInstance(deserialized);
+                return ret;
+            }
+            {{/discriminator}}
+            {{#anyOf}}
+            // deserialzie {{{.}}}
```
Comment (jfeltesse-mdsol) [inline modules/openapi-generator/src/main/resources/Java/libraries/native/anyof_model.mustache:72]: typo

File: modules/openapi-generator/src/main/resources/Java/libraries/native/anyof_model.mustache [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        }
+    }
+
+    // store a list of schema names defined in anyOf
+    public final static Map<String, GenericType> schemas = new HashMap<String, GenericType>();
+
+    public {{classname}}() {
+        super("anyOf", {{#isNullable}}Boolean.TRUE{{/isNullable}}{{^isNullable}}Boolean.FALSE{{/isNullable}});
+    }
+{{> libraries/jersey2/additional_properties }}
```
Comment (jfeltesse-mdsol) [inline modules/openapi-generator/src/main/resources/Java/libraries/native/anyof_model.mustache:107]: jersey2 reference here?

File: modules/openapi-generator/src/main/resources/Java/libraries/native/JSON.mustache [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    * @param descendants a map of oneOf/anyOf descendants.
+    */
+  public static void registerDescendants(Class<?> modelClass, Map<String, GenericType> descendants) {
+    modelDescendants.put(modelClass, descendants);
+  }
+
+  private static JSON json;
+
+  static
+  {
```
Comment (jfeltesse-mdsol) [inline modules/openapi-generator/src/main/resources/Java/libraries/native/JSON.mustache:252]: elsewhere in the code the curly brace is on the same line as `static`

File: modules/openapi-generator/src/main/resources/Java/libraries/native/model.mustache [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+{{#model}}
+{{#additionalPropertiesType}}
+import java.util.Map;
+import java.util.HashMap;
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+{{/additionalPropertiesType}}
+{{/model}}
+{{/models}}
+{{^supportJava6}}
```
Comment (jfeltesse-mdsol) [inline modules/openapi-generator/src/main/resources/Java/libraries/native/model.mustache:19]: is this needed? unsure what it means actually but native == JDK 11

File: modules/openapi-generator/src/main/resources/Java/libraries/native/oneof_model.mustache [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+        }
+    }
+
+    // store a list of schema names defined in oneOf
+    public final static Map<String, GenericType> schemas = new HashMap<String, GenericType>();
+
+    public {{classname}}() {
+        super("oneOf", {{#isNullable}}Boolean.TRUE{{/isNullable}}{{^isNullable}}Boolean.FALSE{{/isNullable}});
+    }
+{{> libraries/jersey2/additional_properties }}
```
Comment (jfeltesse-mdsol) [inline modules/openapi-generator/src/main/resources/Java/libraries/native/oneof_model.mustache:140]: here as well, jersey2

**Comment by bgong-mdsol** [comment]
@wing328 @sebastien-rosset @cbornet @jeff9finger @bbdouglas @sreeshas @jfiala @lukoyanov @karismann @Zomzog @lwlee2608 @bkabrda @jfeltesse-mdsol @jcarres-mdsol

Can someone help to review this PR? Thanks in advance.

File: bin/configs/java-native-8.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+generatorName: java
+outputDir: samples/openapi3/client/petstore/java/native-java8
+library: native
+inputSpec: modules/openapi-generator/src/test/resources/3_0/java/petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml
+templateDir: modules/openapi-generator/src/main/resources/Java
+additionalProperties:
+  artifactId: petstore-openapi3-native-java8
+  hideGenerationTimestamp: true
+  serverPort: "8082"
+  dateLibrary: java8
```
Comment (wing328) [inline bin/configs/java-native-8.yaml:10]: @bgong-mdsol I think you can remove this option as it makes no difference in the `native` library

File: bin/configs/java-native-8.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+generatorName: java
+outputDir: samples/openapi3/client/petstore/java/native-java8
```
Comment (wing328) [inline bin/configs/java-native-8.yaml:2]: I think `native` requires JDK11+ so I would remove `java8` as it's a bit confusing.

File: bin/configs/java-native-8.yaml (hunk 6 above)
Comment (bgong-mdsol) [reply bin/configs/java-native-8.yaml]: updated as suggested in https://github.com/OpenAPITools/openapi-generator/pull/7263/commits/0ddb2a9754c3085dd0f14da7a9b42982965620fb

File: bin/configs/java-native-8.yaml (hunk 7 above)
Comment (bgong-mdsol) [reply bin/configs/java-native-8.yaml]: updated as suggested in https://github.com/OpenAPITools/openapi-generator/commit/0ddb2a9754c3085dd0f14da7a9b42982965620fb

**Comment by bgong-mdsol** [comment]
Are there anything that I need to address before get this merged? Thanks

**Comment by wing328** [comment]
Can you please resolve the merge conflicts? I'll try to review and merge if no question from me later this week.

File: bin/configs/java-native-8.yaml (hunk 6 above)
Comment (bgong-mdsol) [reply bin/configs/java-native-8.yaml]: > Can you please resolve the merge conflicts? I'll try to review and merge if no question from me later this week.

@wing328 Sure, I am going to resolve the merge conflicts.

**Comment by bgong-mdsol** [comment]
@wing328 This PR is ready for review, thanks a lot

**Comment by wing328** [comment]
Tests in Drone.io CI passed: https://cloud.drone.io/OpenAPITools/openapi-generator/11888

**Review by wing328 — APPROVED** [review]

**Pull request merged**